metadata-ingestion/src/datahub/ingestion/run/pipeline_config.py

```diff
@@ -45,6 +45,7 @@ class PipelineConfig(ConfigModel):
     source: SourceConfig
     sink: DynamicTypedConfig
     transformers: Optional[List[DynamicTypedConfig]]
+    flags: Optional[Dict[str, bool]] = None
```

Comment:
flags should be a strongly typed config object, not a dict

it should also have a real default, not `None`

we should also add a comment saying that this is an experimental feature and we don't guarantee backwards compat on it

also, I'm thinking that maybe this belongs under SourceConfig and not at the PipelineConfig level

Reply:
If we can just drop backwards compatibility guarantees that's super nice.
Do we not think we could have sink-related or transformer-related or pipeline-related flags? I like using the most generic version possible, until we have good reason to do differently